lib/core/tools/text_tool/bounding_box.dart

```diff
@@ -0,0 +1,158 @@
+      (topLeft.dy + topRight.dy + bottomLeft.dy + bottomRight.dy) / 4;
+
+  double get width => (topRight.dx - topLeft.dx).abs();
+  double get height => (bottomLeft.dy - topLeft.dy).abs();
+
+  double get activeCornerDirection {
+    switch (activeCorner) {
+      case BoundingBoxCorner.topLeft:
+        return (topLeft - center).direction;
+      case BoundingBoxCorner.topRight:
+        return (topRight - center).direction;
+      case BoundingBoxCorner.bottomLeft:
+        return (bottomLeft - center).direction;
+      case BoundingBoxCorner.bottomRight:
+        return (bottomRight - center).direction;
+      case BoundingBoxCorner.none:
+        return 0;
+    }
+  }
+
+  void setActiveCorner(Offset point, {bool isRotating = false}) {
+    if (point.isWithinRadius(topLeft, boundingBoxCornerRadius)) {
+      activeCorner = BoundingBoxCorner.topLeft;
+    } else if (point.isWithinRadius(topRight, boundingBoxCornerRadius)) {
+      activeCorner = BoundingBoxCorner.topRight;
+    } else if (point.isWithinRadius(bottomLeft, boundingBoxCornerRadius)) {
+      activeCorner = BoundingBoxCorner.bottomLeft;
+    } else if (point.isWithinRadius(bottomRight, boundingBoxCornerRadius)) {
+      activeCorner = BoundingBoxCorner.bottomRight;
+    } else {
+      if (isRotating) {
+        moveCenter(point);
+      } else {
+        update(point);
+      }
+    }
+  }
+
+  void resetActiveCorner() => activeCorner = BoundingBoxCorner.none;
+
+  void update(Offset point, {bool isRotating = false}) {
+    if (activeCorner == BoundingBoxCorner.none) {
+      moveCenter(point);
+    } else if (isRotating) {
+      rotate(point);
+    } else {
+      transform(point);
+    }
+  }
+
+  void updateCorners(
+    Offset topLeft,
+    Offset topRight,
+    Offset bottomLeft,
+    Offset bottomRight, {
+    Offset offset = Offset.zero,
+  }) {
+    this.topLeft = topLeft + offset;
+    this.topRight = topRight + offset;
+    this.bottomLeft = bottomLeft + offset;
+    this.bottomRight = bottomRight + offset;
+  }
+
+  void transform(Offset point) {
+    switch (activeCorner) {
+      case BoundingBoxCorner.topLeft:
+        topLeft = point;
+        bottomLeft = Offset(topLeft.dx, bottomRight.dy);
+        topRight = Offset(bottomRight.dx, topLeft.dy);
+        break;
+      case BoundingBoxCorner.topRight:
+        topRight = point;
+        bottomRight = Offset(topRight.dx, bottomLeft.dy);
+        topLeft = Offset(bottomLeft.dx, topRight.dy);
+        break;
+      case BoundingBoxCorner.bottomLeft:
+        bottomLeft = point;
+        topLeft = Offset(bottomLeft.dx, topRight.dy);
+        bottomRight = Offset(topRight.dx, bottomLeft.dy);
+        break;
+      case BoundingBoxCorner.bottomRight:
+        bottomRight = point;
+        topRight = Offset(bottomRight.dx, topLeft.dy);
+        bottomLeft = Offset(topLeft.dx, bottomRight.dy);
+        break;
+      case BoundingBoxCorner.none:
+        break;
+    }
+  }
+
+  void rotate(Offset point) {}
```

Comment:
Pretty sure rotation should be supported by text tool. If this is a technical limitation of your current implementation it should be disabled in some other way, but definetly implemented before merge.